domain-check.bash

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+# This is meant to be an interim script to (1) evaluate we have
+# enough capabilities inside of the urlgetter pseudo-test and (2)
+# prototype the next generation Web Connectivity.
+#
+# Very alpha. Very basic. Needs much more love.
+#
+# Inspired by great chats and great code. Main source of ispiration is
+# currently https://github.com/Jigsaw-Code/net-analysis/blob/master/netanalysis/blocktest/measure.sh
+
+function fatal() {
+  echo "FATAL: $@" 1>&2
+  exit 1
+}
+
+if [ $# -ne 1 ]; then
+  fatal "usage: $0 domain"
+fi
+domain=$1
+
+function require() {
+  if ! [ -x "$(command -v $1)" ]; then
+    fatal "$1 not installed, please run: $2"
+  fi
+}
+
+require jq "sudo apt install jq"
```

Comment:
I recommend you put all statements that are executed inside a `main()` function, so it's easier to read and understand. Then you call `main "$@"` at the bottom.

Reply:
ACK